test/nbrowser/CopyWithHeaders.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * Test for copying Grist data with headers.
+ */
+
+import {assert, driver, Key} from 'mocha-webdriver';
+import * as gu from 'test/nbrowser/gristUtils';
+import {setupTestSuite} from 'test/nbrowser/testUtils';
+import {createDummyTextArea, removeDummyTextArea} from 'test/nbrowser/CopyPaste';
+
+describe("CopyWithHeaders", function() {
+  this.timeout(90000);
+  const cleanup = setupTestSuite();
+  const clipboard = gu.getLockableClipboard();
+  afterEach(() => gu.checkForErrors());
+  gu.bigScreen();
+
+  after(async function() {
+    await driver.executeScript(removeDummyTextArea);
+  });
+
+  it('should copy headers', async function() {
+    const session = await gu.session().teamSite.login();
+    await session.tempDoc(cleanup, 'Hello.grist');
```

Comment:
Can you add a header if invalid characters that should be encoded in csv ? like:
```
2 " is \this 'ok?
```

Reply:
What is the desired behavior here? I thought the output is just tab-separated fields. In which case the only escaping that might be required is if there's a tab in a column label. That seems to work, probably because all whitespace is turned into a space somewhere else.

Reply:
Ok, I read the code of the `tsvEncode` and it looks alright.